app/src/main/java/ch/epfl/qedit/backend/location/MockLocService.java

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.qedit.backend.location;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import java.util.HashSet;
+import java.util.Set;
+
+/** This class is used to write tests for the parts of the app that need the location service. */
+public class MockLocService implements LocationService {
+    /** This is the id of the mock location provider */
+    public static final String LOCATION_PROVIDER = "mock_provider";
+
+    // This is the set of listeners that need to be updated.
+    private final Set<LocationListener> listeners;
+
+    // This indicates whether the user has permissions to access this service.
+    // Note that this is for test purposes. It allows to test how the UI reacts when the user
+    // doesn't have the access to the location.
+    private boolean permission;
+
+    public MockLocService(Context context) {
+        listeners = new HashSet<>();
+        permission = false;
+    }
+
+    @Override
+    public boolean subscribe(LocationListener listener, int interval) {
+        // In the mock, we will not take into account the interval, because the goal
+        // is just to test how the UI responds to the location updates.
```

Comment:
The more I look at it and the more I think that function is odd. First `interval` is not used. Second if you want to subscribe. You have to give permission. But then if someone else wants to subscribe and you don't want so you have to call `setPermission`

So it would be something like
`setPermission(True)`
`subscribe(...)`
`setPermission(False)`
`subscribe(...)`

Oh wait... why would you want to subscribe someone who you know wouldn't have permission.

Reply:
Ok, let me clarify what permission is. In the real location service, we will need to request the user for the permission to use the location services. I modelled that by returning a boolean if the operation was allowed. That way, in the activities that use the location service, I can request the permission if I see that this method returned false.
Now, in the mock version, I had to artificially model permissions using a boolean flag. That way, I will be able to see how the app reacts if I disable permissions for example.